File: AndroidApp/MultiSportTrainer-Mobile/app/src/main/java/com/example/multisporttrainer/models/SaveRouteRequest.java
package com.example.multisporttrainer.models;

import java.util.List;

public class SaveRouteRequest {

    private int userId;
    private int sessionId;
    private String routeType;
    private List<Integer> coneSequence;

    public SaveRouteRequest(
            int userId,
            int sessionId,
            String routeType,
            List<Integer> coneSequence
    ) {
        this.userId = userId;
        this.sessionId = sessionId;
        this.routeType = routeType;
        this.coneSequence = coneSequence;
    }
}

File: Backend/MultiSportTrainerAPI/MultiSportTrainerAPI/DTOs/SaveRouteRequest.cs
namespace MultiSportTrainerAPI.DTOs
{
    public class SaveRouteRequest
    {
        public int UserId { get; set; }

        public int? SessionId { get; set; }

        public string RouteType { get; set; } = string.Empty;

        public List<int> ConeSequence { get; set; } = new List<int>();
    }
}